app/views/posts/index.html.erb

```diff
@@ -1 +1,21 @@
-<h1>This is the index action for post controller</h1>
\ No newline at end of file
+<div class="container">
+<div class="image"><img src="<%= @user.photo%>" alt="<%= @user.name%>"></div>
+<div class="card">
+<h1 class="header"><%=@user.name %></h1>
+<p class="footer">Number of posts: <%=@user.posts_counter %></p>
+</div>
+<% @posts.each do |post| %>
+<div class="post">
+<p><%=post.title %></p>
+<p><%=post.text %></p>
+<p class="footer">Comments: <%=post.comments_counter %>,
+Likes: <%=post.likes_counter %></p>
+</div>
+<div>
+<% post.recent_comments.each do |comment| %>
+ <div class="bio"><%= comment.author.name %>:  <%= comment.text %></div>
+<% end %>
+</div>
+<% end %>
+<button class="btn"><a href="">Pagination</a></button>
+<div>
```

Comment:
## POST.RB FILE


- Hey, your app throws an error when trying to create a post for a user this is happening as you're calling this [method](https://github.com/FaranoshAmini/rails-blog/blob/6ebf6c188e4ca1f5f8043fc41bed2b460729c895/app/models/post.rb#L7) that does not exist in your `post.rb` file.


- To fix this, kindly create a private method in your `post.rb` file model to increment the post counter

```ruby
private
  def post_counter_update
    author.increment!(:posts_counter) // You can implement this anyway you want.
  end
```
*This should fix the error*

> **[OPTIONAL]** Also, kindly add some data in your seeds file so you can just populate the database with this data using the `rails db:seed` command.